src/equations/covariant_shallow_water_split.jl

```diff
@@ -112,21 +111,45 @@ end
     Gcon_ll = metric_contravariant(aux_vars_ll, equations)
     Gcov_rr = metric_covariant(aux_vars_rr, equations)
     J_ll = area_element(aux_vars_ll, equations)
+    h_s_jump = bottom_topography(aux_vars_rr, equations) -
+               bottom_topography(aux_vars_ll, equations)
 
     # Physical variables
     h_ll = waterheight(u_ll, equations)
     h_rr = waterheight(u_rr, equations)
     h_vcon_ll = momentum_contravariant(u_ll, equations)
     vcon_rr = velocity_contravariant(u_rr, equations)
+    vcov_rr = Gcov_rr * vcon_rr
 
-    # Nonconservative momentum term, consisting of curvature correction and pressure term
-    momentum_noncons = J_ll * (0.5f0 * h_vcon_ll[orientation] *
-                        (Gcon_ll * Gcov_rr * vcon_rr - vcon_rr) +
-                        Gcon_ll[:, orientation] * equations.gravity * h_ll * h_rr)
+    geometric_term = 0.5f0 * h_vcon_ll[orientation] * (Gcon_ll * vcov_rr - vcon_rr)
+    pressure_term = equations.gravity * Gcon_ll[:, orientation] * h_ll *
+                    (h_rr + h_s_jump)
 
-    return SVector(zero(eltype(u_ll)), momentum_noncons[1], momentum_noncons[2])
+    return SVector(zero(eltype(u_ll)), J_ll * (geometric_term[1] + pressure_term[1]),
+                   J_ll * (geometric_term[2] + pressure_term[2]))
 end
 
+# For smooth bottom topography, we can significantly simplify the nonconservative surface
+# term, such that only the pressure term remains.
+@inline function flux_nonconservative_surface_simplified(u_ll, u_rr, aux_vars_ll,
+                                                         aux_vars_rr,
+                                                         orientation::Integer,
+                                                         equations::SplitCovariantShallowWaterEquations2D)
+    # Geometric variables
+    Gcon_ll = metric_contravariant(aux_vars_ll, equations)
+    J_ll = area_element(aux_vars_ll, equations)
+
+    # Physical variables
+    h_ll = waterheight(u_ll, equations)
+    h_rr = waterheight(u_rr, equations)
+
+    pressure_term = equations.gravity * Gcon_ll[:, orientation] * h_ll * h_rr
+
+    return SVector(zero(eltype(u_ll)), J_ll * pressure_term[1], J_ll * pressure_term[2])
+end
```

Comment:
I understand that this is equivalent to `flux_nonconservative_ec` for smooth bottom topographies (continuous across element interfaces). I suggest to state this more clearly in the comments (maybe a docstring would be useful) and the elixirs, and to make a copy of one of the tests (for example [this one](https://github.com/trixi-framework/TrixiAtmo.jl/pull/73/files#diff-87d10937ca36bcc459b4cd08fb7ae058e3c2ee55cc83e76152083d2c1b49b29cR40-R56)), but using `flux_nonconservative_ec` in the surface.